File: Logic Building Problems/Basic Problems/Sum of Squares of Naturals/C/naive_approach.c
/*
Naive Approach - Adding One By One - O(n) Time and O(1) Space
The idea for this naive approach is to run a loop from 1 to n and sum up all the squares.
*/

#include <stdio.h>

int sum_of_squares(int number) {
    // Return the sum of the squared numbers from 1 to n

    int sum_of_squares = 0;
    
    for (int i = 0; i <= number; i++)
        sum_of_squares += i * i;

    return sum_of_squares;
}


int main() {
    int number = 3;

    printf("%d", sum_of_squares(number));

    return 0;
}

File: Logic Building Problems/Basic Problems/Sum of Squares of Naturals/Python/naive_approach.py
"""
Naive Approach - Adding One By One - O(n) Time and O(1) Space
The idea for this naive approach is to run a loop from 1 to n and sum up all the squares.
"""

def sum_of_squares(number: int) -> int:
    """Return the sum of the squared numbers from 1 to n"""
    return sum([i ** 2 for i in range(1, number + 1)])


if __name__ == "__main__":
    number: int = 2
    print(sum_of_squares(number))
